drivers/scheduler/k8s/k8s.go

```diff
@@ -3182,6 +3192,71 @@ func (k *K8s) GetVolumes(ctx *scheduler.Context) ([]*volume.Volume, error) {
 	return vols, nil
 }
 
+// GetPureVolumes  Get the Pure volumes (if enabled) by type (PureFile or PureBlock)
+//
+func (k *K8s) GetPureVolumes(ctx *scheduler.Context, pureVolType string) ([]*volume.Volume, error) {
+	k8sOps := k8sApps
+	var vols []*volume.Volume
+	for _, specObj := range ctx.App.SpecList {
+		if obj, ok := specObj.(*corev1.PersistentVolumeClaim); ok {
+			pvcObj, err := k8sCore.GetPersistentVolumeClaim(obj.Name, obj.Namespace)
+			if err != nil {
+				return nil, err
+			}
+			shouldAdd, err := k.filterPureVolumesIfEnabledByType(pvcObj, pureVolType)
+			if err != nil {
+				return nil, err
+			}
+			if !shouldAdd {
+				continue
+			}
+
+			pvcSizeObj := pvcObj.Spec.Resources.Requests[corev1.ResourceStorage]
+			pvcSize, _ := pvcSizeObj.AsInt64()
+			vol := &volume.Volume{
+				ID:          string(pvcObj.Spec.VolumeName),
+				Name:        obj.Name,
+				Namespace:   obj.Namespace,
+				Shared:      k.isPVCShared(obj),
+				Annotations: make(map[string]string),
+				Labels:      pvcObj.Labels,
+				Size:        uint64(pvcSize),
+			}
+			for key, val := range obj.Annotations {
+				vol.Annotations[key] = val
+			}
+			vols = append(vols, vol)
+		} else if obj, ok := specObj.(*appsapi.StatefulSet); ok {
```

Comment:
Why we need to check it's a statefulset app? What if it's a pod or demonSet deployment?

Reply:
as for as I know its mostly statefulset if not all? this is the same check we do for all functions of this kind actually.

Reply:
What happens if your app is a deamonset or deployment? Will this function return pure volumes for the app?

Reply:
I'm honestly not sure, @Pure-AdamuKaapan do you know more about this?

Reply:
@dsupure this is only getting storage objects: the reason we have to get StatefulSets is because they have the PVC template built into them. Deployments and DaemonSets would use a separate PVC and would be caught by the first if check.

Like @david-enli said, this kind of check is present through most places we list volumes in Torpedo